internal/services/azurestackhci/stack_hci_network_interface_resource.go

```diff
@@ -0,0 +1,351 @@
+package azurestackhci
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/tags"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/azurestackhci/2024-01-01/logicalnetworks"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/azurestackhci/2024-01-01/networkinterfaces"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/extendedlocation/2021-08-15/customlocations"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var (
+	_ sdk.Resource           = StackHCINetworkInterfaceResource{}
+	_ sdk.ResourceWithUpdate = StackHCINetworkInterfaceResource{}
+)
+
+type StackHCINetworkInterfaceResource struct{}
+
+func (StackHCINetworkInterfaceResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return networkinterfaces.ValidateNetworkInterfaceID
+}
+
+func (StackHCINetworkInterfaceResource) ResourceType() string {
+	return "azurerm_stack_hci_network_interface"
+}
+
+func (StackHCINetworkInterfaceResource) ModelObject() interface{} {
+	return &StackHCINetworkInterfaceResourceModel{}
+}
+
+type StackHCINetworkInterfaceResourceModel struct {
+	Name              string                         `tfschema:"name"`
+	ResourceGroupName string                         `tfschema:"resource_group_name"`
+	Location          string                         `tfschema:"location"`
+	CustomLocationId  string                         `tfschema:"custom_location_id"`
+	DNSServers        []string                       `tfschema:"dns_servers"`
+	IPConfiguration   []StackHCIIPConfigurationModel `tfschema:"ip_configuration"`
+	MACAddress        string                         `tfschema:"mac_address"`
+	Tags              map[string]interface{}         `tfschema:"tags"`
+}
+
+type StackHCIIPConfigurationModel struct {
+	Gateway          string `tfschema:"gateway"`
+	PrefixLength     string `tfschema:"prefix_length"`
+	PrivateIPAddress string `tfschema:"private_ip_address"`
+	SubnetID         string `tfschema:"subnet_id"`
+}
+
+func (StackHCINetworkInterfaceResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[a-zA-Z0-9][\-\.\_a-zA-Z0-9]{0,78}[a-zA-Z0-9]$`),
+				"name must be between 2 and 80 characters and can only contain alphanumberic characters, hyphens(-), periods(.) and underlines(_). It must begin and end with an alphanumberic character.",
```

Comment:
@teowa, sorry I forgot to add the description to my review. I took a look a think this mostly LGTM. If we can get the validation function error message fixed up I think this would be GTG! 🚀

Reply:
Thanks for reviewing this, the code is updated, please kindly take another look.